### PR TITLE
rustplayer: init at unstable-2022-12-29

### PR DESCRIPTION
**Pull request by oluceps**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Zaechus — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden

##### Possible improvements
- pname should probably be all lowercase: "rustplayer"
- in `fetchFromGitHub`, `rev` should be after `owner` and `repo`
- A more grammatically correct description would be: "A local audio player and network m3u8 radio player using a terminal interface". The text should also be inside a pair of double quotes ("").
- meta.platforms is needed, not sure what it should be set to though, maybe `platforms.unix`?

##### Comments

**Review by Zaechus — APPROVED** [review]
:+1:

File: pkgs/applications/audio/rustplayer/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  pname = "rustplayer";
+  version = "1.1.0-3";
+
+  src = fetchFromGitHub {
+    owner = "Kingtous";
+    repo = "RustPlayer";
+    rev = "v${version}";
+    sha256 = "sha256-zDkDF/aopmWhK4CLqAC4n1qLLhX3EyXr/MfpUIRYh98=";
+  };
+  LIBCLANG_PATH = "${llvmPackages_latest.libclang.lib}/lib";
```
Comment (ocfox) [inline pkgs/applications/audio/rustplayer/default.nix:21]: Try to use rustPlatform.bindgenHook?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12890,6 +12890,8 @@ with pkgs;
 
   rustfilt = callPackage ../development/tools/rust/rustfilt { };
 
+  rustplayer = callPackage ../applications/audio/rustplayer { };
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:12893]: Duplicate callPackage

**Review by ocfox — APPROVED** [review]

**Review by NickCao — CHANGES_REQUESTED** [review]
build is failing on ofborg.

**Comment by ocfox** [comment]
Possibly missing libva

**Comment by oluceps** [comment]
It is updated to follow the master branch since there is a patch that can fix weird build failures.

**Comment by oluceps** [comment]
@ofborg build

**Review by NickCao — APPROVED** [review]

**Pull request merged**